guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectSearchService.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.common.services.project.service;
+
+import java.util.Collection;
+
+import org.guvnor.common.services.project.model.Project;
+import org.jboss.errai.bus.server.annotations.Remote;
+
+/**
+ * {@link Project} remote search interface
+ */
+@Remote
+public interface ProjectSearchService {
```

Comment:
I totally agree.

This is an initial implementation that works fairly well. At least for an small number of instances.

AFAIK @mrietveld is working on index based implementations for searching OUs, repos & projects. He plans to apply this new implementation once this PR is merged on master.

Aside from that, I also agree some context is required, like the repo user wants to manipulate its permissions, so that only that repo's project are fetched. Right now, project permissions are at the same level that OUs or repos but it is possible to add them as subnodes of each repo.

Anyway, I'd wait for @mrietveld implementation to decide what the best strategy.